### PR TITLE
Proposal: urlsFromPrember

### PR DESCRIPTION
**Pull request by rwwagner90**
This would allow passing not only `urlsForPrember`, but also for prember to output `urlsFromPrember`. The code and the API might not be ideal, but I have had a need to get the urls from prember for quite awhile, so it would be great if we could get this or something like it in!

It doesn't work ideally because it does not seem to be possible to run addons `after` prember. Possibly because prember's build is async?

Ideally, we could add this in the addon:
```json
 "ember-addon": {
    "configPath": "tests/dummy/config",
    "after": [
      "prember"
    ]
  }
```
It seems to have no effect though, as my `treeForPublic` still executed before prember did its build.

### CODE REVIEW

File: lib/prerender.js [hunk 1]
```diff
@@ -40,7 +40,15 @@ class Prerender extends Plugin {
     }
 
     for (let plugin of this.plugins) {
-      this.urls = this.urls.concat(await plugin(this.inputPaths[0], visit));
+      if (plugin.urlsForPrember) {
```
Comment (villander) [inline lib/prerender.js]: we need of tests here

File: index.js [hunk 2]
```diff
@@ -40,6 +40,20 @@ function loadPremberPlugins(context) {
 
   return addons
     .filter((addon) => addon.pkg.keywords.includes('prember-plugin'))
-    .filter((addon) => typeof addon.urlsForPrember === 'function')
-    .map((addon) => addon.urlsForPrember.bind(addon));
+    .filter((addon) => {
```
Comment (villander) [inline index.js]: we need of tests here too

**Comment by rwwagner90** [comment]
@villander there do not seem to be any tests around the existing `urlsForPrember`, so I have no idea how to write tests for `urlsFromPrember`.

We should also document this new functionality, which I can definitely do.

**Comment by villander** [comment]
@rwwagner90 sorry for delay. We don't have coverage on `postprocessTree`. But you can create tests like this on `ember-asset-loader`: 

https://github.com/ember-engines/ember-asset-loader/blob/ae615df700b20fd619be76ecb94e7699d96d0cba/node-tests/manifest-generator-test.js#L75

**Comment by rwwagner90** [comment]
@villander no worries! I'm just not sure why we need tests, since none previously existed? If you guys could add an existing one for `urlsForPrember`, I would be happy to try to mimic it and add one for `urlsFromPrember`.

**Comment by villander** [comment]
Makes sense @rwwagner90. Tests are always welcome, but we don't have any tests about it previously. 

Can you provide a example this working in a public repo, and add a documentation please? with this, I can approve and add tests after merge.

**Comment by rwwagner90** [comment]
@villander I used this for some client work, so I don't have it in a public repo, but I can work on a public one and docs, for sure.

**Comment by rwwagner90** [comment]
@villander I'm unlikely to create a public repo using this for awhile. Can we go ahead and merge it in please?

**Comment by villander** [comment]
@rwwagner90 can you add documentation for it please?

**Comment by rwwagner90** [comment]
@villander just added some docs. Hopefully they work for you!

**Comment by villander** [comment]
Thanks @rwwagner90

**Comment by villander** [comment]
v1.0.4 publish @rwwagner90, sorry for the delay - https://www.npmjs.com/package/prember
